File: hw/Models/Commitment.swift
import Foundation

struct Commitment: Codable {
    enum CodingKeys: String, CodingKey {
        case assetId = "asset_id"
        case value = "value"
        case abf = "abf"
        case vbf = "vbf"
        case assetGenerator = "asset_generator"
        case valueCommitment = "value_commitment"
        case blindingKey = "blinding_key"
    }
    let assetId: Data?
    let value: UInt64?
    let abf: Data?
    let vbf: Data?
    var assetGenerator: Data?
    var valueCommitment: Data?
    var blindingKey: Data?
}

File: jade/src/commonMain/kotlin/com/blockstream/jade/api/Commitment.kt
package com.blockstream.jade.api

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Commitment(
    @SerialName("asset_id")
    val assetId: ByteArray,
    val value: Long?,
    val abf: ByteArray?,
    val vbf: ByteArray?,
    @SerialName("blinding_key")
    val blindingKey: ByteArray
) : JadeSerializer<Commitment>() {
    override fun kSerializer() = serializer()
    override fun encodeDefaultsValues() = false
}
